### PR TITLE
Initialise RSTensor proto

### PR DESCRIPTION
**Pull request by kamathhrishi**
## Description
This pull request aims to add protobuf files for newly created [ReplicatedSharedTensor](https://github.com/OpenMined/SyMPC/blob/main/src/sympc/tensor/replicatedshare_tensor.py) in SyMPC. 
 I have also added RSTensor to SyMPC API https://github.com/OpenMined/SyMPC/pull/185. 

## How has this been tested?
https://github.com/OpenMined/SyMPC/pull/207 has the tests for this PR. 

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by kamathhrishi** [comment]
I don't think I can request a review since I am not a maintainer. 

@gmuraru, @LaRiffle, @aanurraj, and @madhavajay have a look at the PR and let me know if I am going about the right way. If I have missed out on anything.

**Comment by kamathhrishi** [comment]
> I don't think I can request a review since I am not a maintainer.
> 
> @gmuraru, @LaRiffle, @aanurraj, and @madhavajay have a look at the PR and let me know if I am going about the right way. If I have missed out on anything.

I have sorry, I realized there are some changes I need to make before a proper review.

File: packages/syft/src/syft/lib/sympc/rst_share.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+def object2proto(obj: object) -> ReplicatedSharedTensor_PB:
+    share: ReplicatedSharedTensor = obj
+
+    session = protobuf_session_serializer(share.session)
+    proto = ReplicatedSharedTensor_PB(session=session)
+
+    tensor_proto_list = []
+
+    for tensor in share.shares:
+        tensor_proto_list.append(protobuf_tensor_serializer(tensor))
```
Comment (gmuraru) [inline packages/syft/src/syft/lib/sympc/rst_share.py:24]: Q: Why not directly?
`proto.tensor.append(protobuf_tensor_serializer(tensor))`?

File: packages/syft/src/syft/lib/sympc/rst_share.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    return proto
+
+
+def proto2object(proto: ReplicatedSharedTensor_PB) -> ReplicatedSharedTensor:
+    session = protobuf_session_deserializer(proto=proto.session)
+
+    output_shares = []
+
+    for tensor in proto.tensor:
+
```
Comment (gmuraru) [inline packages/syft/src/syft/lib/sympc/rst_share.py:39]: MAYBE: No need for empty space between for and the other line

**Review by gmuraru — CHANGES_REQUESTED** [review]
Check comments

File: packages/syft/src/syft/lib/sympc/rst_share.py (hunk 1 above)
Comment (rasswanth-s) [reply packages/syft/src/syft/lib/sympc/rst_share.py:24]: George ,I think repeated message(TensorData) do not have the append attribute, it is available for the builtin types(string,int32 etc) but not for repeated message attributes. They have add,extend attributes for addition of elements.

File: packages/syft/src/syft/lib/sympc/rst_share.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    session = protobuf_session_serializer(share.session)
+    proto = ReplicatedSharedTensor_PB(session=session)
+
+    tensor_proto_list = []
+
+    for tensor in share.shares:
+        tensor_proto_list.append(protobuf_tensor_serializer(tensor))
+
+    if len(tensor_proto_list) != 0:
```
Comment (rasswanth-s) [inline packages/syft/src/syft/lib/sympc/rst_share.py:26]: MAYBE : We could do directly , `proto.tensor.extend(tensor_proto_list)`

File: packages/syft/src/syft/lib/sympc/rst_share.py (hunk 1 above)
Comment (gmuraru) [reply packages/syft/src/syft/lib/sympc/rst_share.py:24]: Then, couldn't we use:
```
proto.tensor.extend(protobuf_tensor_serializer(tensor))
```
?

Comment (kamathhrishi) [reply packages/syft/src/syft/lib/sympc/rst_share.py:24]: You were right @gmuraru. Its possible, I will make the changes.

File: packages/syft/src/syft/lib/sympc/rst_share.py (hunk 3 above)
Comment (kamathhrishi) [reply packages/syft/src/syft/lib/sympc/rst_share.py:26]: Yep, it is possible. Thanks.

**Comment by kamathhrishi** [comment]
@gmuraru and @rasswanth-s I have made the relevant changes. You can re-review it now.

File: packages/syft/src/syft/lib/sympc/rst_share.py (hunk 1 above)
Comment (rasswanth-s) [reply packages/syft/src/syft/lib/sympc/rst_share.py]: Yeah @gmuraru the append method works,I think I saw it in a  older documentation of protobuf that append does not work.

File: packages/syft/proto/lib/sympc/replicatedshared_tensor.proto [hunk 4]
```diff
@@ -0,0 +1,11 @@
+syntax = "proto3";
+
+package syft.lib.sympc;
+
+import "proto/lib/torch/tensor.proto";
+import "proto/lib/sympc/session.proto";
+
+message ReplicatedSharedTensor {
+  repeated syft.lib.torch.TensorData tensor = 1;
+  syft.lib.sympc.MPCSession session = 2;
```
Comment (rasswanth-s) [inline packages/syft/proto/lib/sympc/replicatedshared_tensor.proto:10]: @gmuraru Should we also modify the session attribute to include uuid and config.
Similar to the one in your [Current PR](https://github.com/OpenMined/PySyft/blob/cbdf5088bf59d654d11a6594937a9e36e92a9e15/packages/syft/proto/lib/sympc/share_tensor.proto)

Comment (kamathhrishi) [reply packages/syft/proto/lib/sympc/replicatedshared_tensor.proto:10]: No need. The session serialization is an independent component. RST uses session serializer and deserializer so it is taken care of without having to modify anything here.

Comment (kamathhrishi) [reply packages/syft/proto/lib/sympc/replicatedshared_tensor.proto:10]: Good suggestion.

**Review by rasswanth-s — APPROVED** [review]
Looks good to me.

**Review by gmuraru — APPROVED** [review]

**Pull request merged**